pkg/network/dns_snooper.go

```diff
@@ -197,16 +188,6 @@ func (s *SocketFilterSnooper) pollPackets() {
 		default:
 		}
 
-		if err == nil {
-			s.processPacket(data, captureInfo.Timestamp)
-			continue
-		}
-
-		// Immediately retry for EAGAIN
-		if err == syscall.EAGAIN {
-			continue
-		}
-
 		// Sleep briefly and try again
 		time.Sleep(5 * time.Millisecond)
```

Comment:
I know this isn't new code, but I'm curious how this time delay was arrived at? Does it impact our ability to keep up with high-throughput DNS traffic?

Reply:
I think this was arbitrarily chosen, but tests don't show us dropping any packets. 

Looking closer at this code, I'm realizing that the behavior changes slightly in this PR. We used to keep reading without a sleep between reads until ZeroCopyReadPacket   returned a timeout. But VisitPacket will only read one packet, and then sleep 5 milliseconds. That means that a burst of DNS traffic is okay as long as it all fits in the circular buffer allocated by afpacket. 

I'll refactor VisitPacket to be VisitPackets, and drain the buffer in a similar fashion.